module/vision/Yolo/src/Yolo.cpp

```diff
@@ -0,0 +1,317 @@
+#include "Yolo.hpp"
+
+#include <chrono>
+#include <getopt.h>
+#include <iostream>
+#include <vector>
+
+#include "extension/Configuration.hpp"
+
+#include "message/input/Image.hpp"
+#include "message/vision/Ball.hpp"
+#include "message/vision/BoundingBoxes.hpp"
+#include "message/vision/FieldIntersections.hpp"
+#include "message/vision/Goal.hpp"
+#include "message/vision/Robot.hpp"
+
+#include "utility/math/coordinates.hpp"
+#include "utility/support/yaml_expression.hpp"
+#include "utility/vision/Vision.hpp"
+#include "utility/vision/fourcc.hpp"
+#include "utility/vision/projection.hpp"
+
+namespace module::vision {
+
+    using extension::Configuration;
+
+    using message::input::Image;
+    using message::vision::Ball;
+    using message::vision::Balls;
+    using message::vision::BoundingBox;
+    using message::vision::BoundingBoxes;
+    using message::vision::FieldIntersection;
+    using message::vision::FieldIntersections;
+    using message::vision::Goal;
+    using message::vision::Goals;
+    using message::vision::Robot;
+    using message::vision::Robots;
+
+    using utility::math::coordinates::cartesianToReciprocalSpherical;
+    using utility::math::coordinates::cartesianToSpherical;
+    using utility::support::Expression;
+    using utility::vision::unproject;
+
+    Yolo::Yolo(std::unique_ptr<NUClear::Environment> environment) : Reactor(std::move(environment)) {
+
+        on<Configuration>("Yolo.yaml").then([this](const Configuration& config) {
+            // Use configuration here from file Yolo.yaml
+            this->log_level                       = config["log_level"].as<NUClear::LogLevel>();
+            cfg.model_path                        = config["model_path"].as<std::string>();
+            cfg.ball_confidence_threshold         = config["ball_confidence_threshold"].as<double>();
+            cfg.goal_confidence_threshold         = config["goal_confidence_threshold"].as<double>();
+            cfg.robot_confidence_threshold        = config["robot_confidence_threshold"].as<double>();
+            cfg.intersection_confidence_threshold = config["intersection_confidence_threshold"].as<double>();
+            cfg.device                            = config["device"].as<std::string>();
+        });
+
+        on<Startup>().then("Load Yolo Model", [this] {
+            compiled_model = core.compile_model(cfg.model_path, cfg.device);
+            infer_request  = compiled_model.create_infer_request();
+        });
```

Comment:
is there a reason this isn't done in Configuration? 
`core` only seems to be used once, here, so I'm wondering if it needs to be a member variable?

Reply:
Good point, idk why I did it in startup